app/src/main/java/org/systers/mentorship/view/fragments/TaskViewModelFactory.kt

```diff
@@ -0,0 +1,15 @@
+package org.systers.mentorship.view.fragments
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import org.systers.mentorship.models.Relationship
+import org.systers.mentorship.viewmodels.TasksViewModel
+
+class TaskViewModelFactory(private val mentorshipRelation: Relationship) : ViewModelProvider.Factory {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+       return TasksViewModel(mentorshipRelation) as T
+    }
+
+}
```

Comment:
Have you created this class for any specific reason? I know that it's used to create a ViewModel but we're already getting the instance of **TasksViewModel** class in **TasksFragment** using **get** method. So, are they any specific reasons for creating this class

Reply:
Initially I used it to initialize TaskViewModel with constructor params, but when I was removing task adding and deleting, I forgot about it.

Reply:
Okay 👍